### DISCUSSION TITLE
`gum confirm` should ESC and ⌃C both exit code 130?

### DISCUSSION THREAD

**Discussion by mxcl** [charmbracelet/gum#263] [Ideas]
Since `gum confirm` is like a GUI dialog and on all *graphical* platforms ESC is interpreted as “pressing cancel” I’m not sure ESC should have an exit code of `130`. Maybe I’m wrong, but in my scripts I'd like to handle ⌃C as “abort” and ESC as “the user pressed cancel”. Then I can do

```sh
case $(gum confirm) in
0) proceed;;
1) skip;;
130) exit 130;;
sac
```

**Comment by maaslalani** [comment]
Personally, (and I may be wrong + open to other opinions) think of <kbd>Esc</kbd> (in gum confirm) as a shortcut to pressing "No" (similar to "n/N") rather than a full cancel option. I probably wouldn't expect everything to cancel as if I had press <kbd>ctrl+c</kbd>, but rather just skip / say "No" to the current prompt.

**Reply by mxcl** [reply]
well indeed, currently `Esc` gives exit code `130`, same as `ctrl+c` and I was expecting it to be `1`. Sorry if I wasn't clear.

I do like that `ctrl+c` now gives `130` though as I can now handle aborts (since 0.9.0)

**Reply by maaslalani** [reply]
Ah sorry, I misread your discussion post, you're absolutely correct and we will change this 👍

**Comment by maaslalani** [comment]
This should be fixed with https://github.com/charmbracelet/gum/commit/36ef76185c49baf998a2ff6c0e8da948410d2579

**Reply by mxcl** [reply]
Terrific thanks!
